### ISSUE TITLE
Feature request: convert master dimension to normal

### ISSUE THREAD

**Issue by Dannypan** [Captain-Chaos/WorldPainter#356]
I’ve seen a few people on Discord now make a master dimension not really knowing what it is and being dismayed when their project isn’t what it is. 

Could an option to convert a master dimension to a normal one be implemented?

**Comment by Captain-Chaos** [comment]
I don't think that is necessary. You can already do that by going to the Surface dimension (by hitting Ctrl+M) and just adding all the tiles (by hitting `Select all tiles` and then `Add or expand tiles`; it may take a long time or even run out of memory though, depending on how large the master dimension is). Then you can delete the master dimension.

**Comment by Dannypan** [comment]
Like you said people are likely to run out of memory and it scales badly. I just did a little test on a 1k map. It's now a 16k map and the one hill I made in the MD is far too big. I can't scale it down perfectly to 1024 x 1024 because the scaling option doesn't go that far down (and rescaling isn't 1:1 perfect).

People are making maps bigger than 1k too. A 4k map on a master dimension is over 65k - that's going to crash a lot of PCs and most people don't want maps that big.

I mentioned some of this on Reddit but I'll repost my suggestions here so it's all in one place:

- Rename it to "Supersize Dimension" or something along those lines.
- After pressing create, it shows popup with a timer on the proceed stating something like:

> You are about to create a supersized dimension. This dimension will be exported at 16 times bigger than your chosen dimension size. Please note that:
> * You will need to add further tiles with CTRL + M to create 1:1 areas for fine detailing.
> * Supersized dimensions are for creating large, undetailed areas between detailed areas for extremely large projects.
> * One pixel in a supersized dimension is equal to one chunk in Minecraft.
> * Your overall project size and file size may be much larger than you intend.
> If you are happy with all of this, press Create. Otherwise, press Cancel.
(I know there's the info box next to the master dimension option when creating a world but a lot of people ignore stuff like that and just rush into WP without taking the time to learn or understand it!)

- Include an option to convert a master dimension to the surface dimension.

**Comment by Captain-Chaos** [comment]
But converting the master dimension to a surface dimension amounts to the same thing, so it would have exactly the same issues. It's pretty much too late at that point.

WorldPainter works pretty hard to show you how big the map is going to be when you make a master dimension. Most of the things you propose it _already_ tells you at some point. You have to completely ignore a lot of information to be caught unawares. There's not really much I can do to for people like that; they are just going to have to learn the hard way. You know what they say: if you try to make something fool-proof, someone will invent a better fool.

Your point about the name is well taken though. Perhaps a more on the nose name would help. I'll consider that.

**Comment by Captain-Chaos** [comment]
I just realised that you probably mean changing the master dimension to a normal dimension one to one, right? Hmmm.

**Comment by Dannypan** [comment]
By 1:1 I mean it becomes sixteen times smaller, so one pixel is one block like normal instead of one pixel is one chunk. I don’t think people who accidentally use MD want (or should be making) maps that are 65k lol
